SentinelHub/sentinelhub/session.py

```diff
@@ -31,13 +31,19 @@ def __init__(self, base_url, client_id, client_secret):
         :param client_secret: A Sentinel Hub Oauth client secret
         :type client_secret: str
         """
-        self.oauth_url = f"{base_url}/oauth/token"
+        self.oauth_url = self.select_oauth_url(base_url)
         self.client_id = client_id
         self.client_secret = client_secret
 
         self._token = None
         _ = self.token
 
+    @staticmethod
+    def select_oauth_url(base_url):
+        if base_url == "https://sh.dataspace.copernicus.eu":
+            return "https://identity.dataspace.copernicus.eu/auth/realms/CDSE/protocol/openid-connect/token"
+        return f"{base_url}/oauth/token"
+
```

Comment:
Just a comment for the future, it could be useful if the endpoints could be selectable from a dropdown menu.

Reply:
Built a zipped package and installed it, seems to work as expected, nice!

![image](https://github.com/sentinel-hub/sentinelhub-qgis-plugin/assets/37869250/5da86c00-a8b9-4ad3-ae5d-7a1147393b47)